tests/test_run_simulation.py

```diff
@@ -556,5 +565,132 @@ def test_user_spec_mu(self):
         self.assertEqual(expected, result)
 
 
+class TestResolveAmbiguities(unittest.TestCase):
+
+    def setUp(self):
+        random.seed(4973)
+
+    def testBadInput(self):
+        seq = "SGDKLJFAHHH"
+        expected = "CGGGLJFAACA"
+        result = resolve_ambiguities(seq)
+        self.assertEqual(expected, result)
```

Comment:
There should also be a check here that, after resolving ambiguities, the resulting sequence is rejected as containing illegal characters